### PR TITLE
[PT FE] Support aten::_transformer_encoder_layer_fwd

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - Add PyTorch FE translation for `aten::_transformer_encoder_layer_fwd`, the fast-path op that `torch.jit.trace` of `torch.nn.TransformerEncoderLayer` produces.
 - Register `aten._transformer_encoder_layer_fwd.default` as well, since `torch.export` in strict mode emits the same op.
 - Move the multi-head attention subgraph out of the `aten::_native_multi_head_attention` translator into a shared `build_multi_head_attention()` helper in `utils`, and build it with `ScaledDotProductAttention` when the attention weights are not requested.

### Tickets:
 - CVS-192051

### AI Assistance:
 - AI assistance used: yes
 - Copilot drafted the translator, the shared helper and the layer tests.

### CODE REVIEW

File: tests/layer_tests/pytorch_tests/test_transformer_encoder_layer.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    def test_transformer_encoder_layer_fwd(self, norm_first, use_gelu, mask_type,
+                                           ie_device, precision, ir_version):
+        mask_data = self._get_mask_data(mask_type)
+        self._test(aten_transformer_encoder_layer_fwd(norm_first, use_gelu, mask_type, mask_data),
+                   "aten::_transformer_encoder_layer_fwd", ie_device, precision, ir_version,
+                   trace_model=True, custom_eps=1e-4)
+
+    @pytest.mark.nightly
+    @pytest.mark.precommit
+    @pytest.mark.parametrize("norm_first", [False, True])
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_transformer_encoder_layer.py]: This test will fail: with `norm_first=True` the fast-path op is never emitted.

```python
@pytest.mark.parametrize("norm_first", [False, True])
```

PyTorch's `TransformerEncoderLayer` disables the fused fast path when `norm_first=True` (`why_not_sparsity_fast_path = "norm_first was True"`). In that case the traced graph decomposes into elementary ops and `aten::_transformer_encoder_layer_fwd` is not present — but `_test()` asserts the op kind exists, so this parametrization fails with `Operation aten::_transformer_encoder_layer_fwd type doesn't exist in provided graph`.

`norm_first=True` is already covered by `test_transformer_encoder_layer_fwd` (which calls the op directly), so restrict this test to `False`:

```suggestion
    @pytest.mark.parametrize("norm_first", [False])
```

File: src/frontends/pytorch/src/utils.cpp [hunk 2]
```diff
@@ -821,6 +827,113 @@ Output<Node> masked_select(const NodeContext& context, const Output<Node>& data,
...
+        weights = context.mark_node(std::make_shared<v1::Multiply>(qk, scale));
+        if (mask_is_boolean) {
+            const auto minus_inf = context.mark_node(
+                v0::Constant::create(element::f32, Shape{}, {-std::numeric_limits<float>::infinity()}));
+            const auto minus_inf_conv = context.mark_node(std::make_shared<v1::ConvertLike>(minus_inf, weights));
+            weights = context.mark_node(std::make_shared<v1::Select>(mask, minus_inf_conv, weights));
+        } else if (mask.get_node_shared_ptr()) {
+            weights = context.mark_node(std::make_shared<v1::Add>(weights, mask));
+        }
+        weights = context.mark_node(std::make_shared<v8::Softmax>(weights, -1));
```
Comment (bumbosiepsak) [inline src/frontends/pytorch/src/utils.cpp:913]: Behavior change for the existing `aten::_native_multi_head_attention` second output, currently untested.

The previous translator returned the **pre-softmax** scaled scores as the second output and averaged with `ReduceSum` + divide-by-heads. This refactor returns **post-softmax** weights and averages with `ReduceMean`. The new result is more correct, but it silently changes the second output of an already-supported op, and the native-MHA layer test only checks output `[0]`, so this path is unverified.

Please confirm the change is intentional and add coverage comparing the second output (both averaged and non-averaged) against the framework.

File: src/frontends/pytorch/src/utils.cpp [hunk 3]
```diff
@@ -821,6 +827,113 @@ Output<Node> masked_select(const NodeContext& context, const Output<Node>& data,
...
+    const auto v = project(value, embed_dim_2x, embed_dim_3x);
+
+    Output<Node> mask = attn_mask;
+    if (mask.get_node_shared_ptr() && mask_type == 1) {
+        // Key padding mask of shape [batch, sequence] has to be made broadcastable to the attention
+        // weights shape [batch, heads, sequence, sequence].
+        const auto axes = context.mark_node(v0::Constant::create(element::i64, Shape{2}, {1, 2}));
+        mask = context.mark_node(std::make_shared<v0::Unsqueeze>(mask, axes));
+    }
+    const bool mask_is_boolean = mask.get_node_shared_ptr() && mask.get_element_type() == element::boolean;
```
Comment (bumbosiepsak) [inline src/frontends/pytorch/src/utils.cpp:887]: Non-boolean masks are now silently accepted for `aten::_native_multi_head_attention` (previous hard error removed).

The old native-MHA translator explicitly rejected non-boolean masks with `PYTORCH_OP_CONVERSION_CHECK(1, "Non-boolean masks are not supported.")`. Dropping the guard is needed for `_transformer_encoder_layer_fwd` (additive float masks), but it also broadens native-MHA behavior for a previously-rejected input. Please confirm this is intended for the native-MHA path as well.

File: tests/layer_tests/pytorch_tests/test_transformer_encoder_layer.py (hunk 1 above)
Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_transformer_encoder_layer.py]: Good catch — fixed. `norm_first=True` disables PyTorch's fused fast path, so the module-level trace never produces `aten::_transformer_encoder_layer_fwd` in that case. Restricted this test's `norm_first` parametrization to `[False]`; the `True` case is already covered by `test_transformer_encoder_layer_fwd`, which invokes the fast-path op directly.

File: src/frontends/pytorch/src/utils.cpp (hunk 2 above)
Comment (mvafin) [reply src/frontends/pytorch/src/utils.cpp:913]: Confirmed intentional, and verified directly against `torch.ops.aten._native_multi_head_attention`: the framework's second output is post-softmax weights (they sum to 1 along the last axis), matching what `build_multi_head_attention()` now produces. The old pre-softmax/`ReduceSum`-based averaging was the actual bug.

Added `test_native_multi_head_attention_weights` covering the second output for both `average_attn_weights=True/False`, comparing against the framework.

File: src/frontends/pytorch/src/utils.cpp (hunk 3 above)
Comment (mvafin) [reply src/frontends/pytorch/src/utils.cpp:887]: Confirmed intentional and correct: I verified directly against real `torch.ops.aten._native_multi_head_attention` that it does accept additive (non-boolean) masks and produces matching results, so dropping the guard is not a broadening of unsupported input — the original assert was overly conservative.

Extended `test_native_multi_head_attention_weights` with a `mask_dtype=["bool", "float"]` parametrization to cover this path against the framework. This also surfaced a pre-existing CPU-plugin defect (unrelated to this PR's FE conversion — the produced graph is mathematically correct) that only triggers when an additive key-padding mask happens to have as many elements as the QK^T MatMul's last dimension; that specific case is marked `xfail` with a detailed explanation referencing `dnnl_postops_composer.cpp:483`.

**Comment by ababushk** [comment]
Please ignore `Category: TF FE` label, and apologies to reviewers - I've added it manually to workaround an issue with GitHub Actions workflow

File: tests/layer_tests/pytorch_tests/test_native_multi_head_attention.py [hunk 4]
```diff
@@ -90,3 +111,37 @@ def _get_mask_data(self, mask):
...
+            # `seq` elements, i.e. the same size as the last dimension of the preceding QK^T MatMul.
+            # The CPU plugin's post-ops fusion (DnnlPostOpsComposer::appendBinary) misidentifies this
+            # Add as a fusable per-output-channel bias and fails with
+            # "Check 'data.size() == OC' failed ... data size: 6 OC: 0" at graph compilation time,
+            # because MatMul (unlike Convolution/FullyConnected) has no well-defined output-channel
+            # axis for this fusion. This is a CPU plugin defect, not a PyTorch FE conversion issue -
+            # the produced graph is mathematically correct and matches PyTorch's output.
+            pytest.xfail("CPU plugin fails to compile Add(MatMul_output, mask) when the additive mask "
+                         "has as many elements as the MatMul's last dimension "
+                         "(dnnl_postops_composer.cpp:483, 'data.size() == OC' assertion)")
```
Comment (bumbosiepsak) [inline tests/layer_tests/pytorch_tests/test_native_multi_head_attention.py:143]: [nitpick] Consider adding explicit tracking information:
```suggestion
                         "(dnnl_postops_composer.cpp:483, 'data.size() == OC' assertion). Tracked in: CVS-XXXXXX")
```

**Review by bumbosiepsak — APPROVED** [review]
One nitpick - consider adding now or in subsequent deliveries.
